src/main/java/revai/StreamingClient.java

```diff
@@ -0,0 +1,205 @@
+   * @see StreamContentType
+   */
+  public StreamingClient(String accessToken, StreamContentType streamContentType) {
+    this.accessToken = accessToken;
+    this.streamContentType = streamContentType;
+    this.client = setClient();
+  }
+
+  /**
+   * This method is used for internal testing of the websocket connection against a mock webserver
+   * and should not be used
+   *
+   * @param scheme
+   */
+  public void setScheme(String scheme) {
+    this.scheme = scheme;
+  }
+
+  public String getHost() {
+    return host;
+  }
+
+  /**
+   * This method is used for internal testing and should not be used
+   *
+   * @param host
+   */
+  public void setHost(String host) {
+    this.host = host;
+  }
+
+  public String getMetadata() {
+    return metadata;
+  }
+
+  /**
+   * Setting metadata is optional but helps to identify the stream.
+   *
+   * @param metadata String size restricted to <= 256 characters
+   */
+  public void setMetadata(String metadata) {
+    this.metadata = metadata;
+  }
+
+  public String getCustomVocabularyId() {
+    return customVocabularyId;
+  }
+
+  /**
+   * Setting a custom vocabulary is optional. This method assigns a custom vocabulary to be used
+   * during the stream. Custom vocabularies are submitted prior to usage in the stream and assigned
+   * an Id.
+   *
+   * @param customVocabularyId
+   */
+  public void setCustomVocabularyId(String customVocabularyId) {
+    this.customVocabularyId = customVocabularyId;
+  }
+
+  public Boolean getFilterProfanity() {
+    return filterProfanity;
+  }
+
+  /**
+   * Setting the profanity filter is optional. More info on the expected can be found at
+   * https://www.rev.ai/docs/streaming#section/WebSocket-Endpoint/Filter-Profanity
+   *
+   * @param filterProfanity This is false by default.
+   */
+  public void setFilterProfanity(Boolean filterProfanity) {
+    this.filterProfanity = filterProfanity;
+  }
+
+  /**
+   * Sends an HTTP request and upon authorization is upgraded to a websocket connection. Use
+   * websocket listener to handle websocket events.
+   *
+   * @param revAiWebsocketListener
+   * @see WebSocketListener
+   * @throws URISyntaxException
+   */
+  public void connect(RevAiWebsocketListener revAiWebsocketListener) throws URISyntaxException {
+    Listener listener = new Listener(revAiWebsocketListener);
+    Request request = new Request.Builder().url(buildURL()).build();
+    webSocket = client.newWebSocket(request, listener);
+  }
+
+  /**
+   * Sends data over websocket in the form of a ByteString
+   *
+   * @param byteString
+   */
+  public void sendAudioData(ByteString byteString) {
+    webSocket.send(byteString);
+  }
+
+  /** Will close the websocket connection by informing the server that it's the End of Stream */
+  public void close() {
+    webSocket.send("EOS");
+  }
+
+  private String buildURL() throws URISyntaxException {
+    URIBuilder uriBuilder = new URIBuilder();
+    if (scheme != null) {
```

Comment:
I agree that it would be but I need this so I can run the unit tests against a fake/mock webserver. This was the simplest way I could find to unit test streaming.

Reply:
but why would the unit tests be an issue. What I was suggesting was 

```
public class StreamingClient {
   private String scheme = "wss";

   public void setScheme(String scheme) {
      this.scheme = scheme;
   }

   private String buildURL() ... {
        uriBuilder.scheme(scheme).host(..)... // etc
    }
}
```

and similar for host. i think this would all still work just fine for your mock webserver, no?
its a very minor point of course.

Reply:
As discussed in slack I removed the `URIBuilder` and instead am using the `HttpUrl.Builder` supplied by OkHttp. The builder silently converts `https` and `http` schemes to `wss` and `ws`. I added the switch statement in case a customer decides to manually set it to `wss` or `ws`.

Reply:
I would do it slightly differently, right now if you do `setScheme("ws")` it wouldnt work, which would be surprising - the fact that we need to use http/https behind the scenes shouldnt be exposed to the caller of our class. Even though its somewhat minor, since we dont normally expect people to change it at all.

simplest way i think would be to validate in the setter. something like this:

```
void setScheme(string scheme) {
    string actualScheme = scheme == "wss" ? "https" : (scheme == "ws" ? "http" : scheme);
    if (actualScheme != "http" && actualScheme != "https") { 
       throw new IllegalArgumentException("invalid scheme...");
    }. 
  
}
```

Reply:
This makes sense. I think conditional expressions can be a little obscure. Do you have any objections to using a switch statement instead? Like this...
```
public void setScheme(String scheme) {
    switch (scheme.toLowerCase()) {
      case "wss":
      case "https":
        this.scheme = "https";
        break;
      case "ws":
      case "http":
        this.scheme = "http";
        break;
      default:
        throw new IllegalArgumentException("Invalid scheme: " + scheme);
    }
  }
```